sql/catalog/descs: TestTxnWithExecutorDataDriven failed

sql/catalog/descs.TestTxnWithExecutorDataDriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9240689?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9240689?buildTab=artifacts#/) on master @ [1d2dff3f5830ba5b8aa7187095b06a4021b9ff85](https://github.com/cockroachdb/cockroach/commits/1d2dff3f5830ba5b8aa7187095b06a4021b9ff85):

Fatal error:

```
panic: test timed out after 55s

```
Stack: 

```
goroutine 34530 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2036 +0x88
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x38
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestTxnWithExecutorDataDriven
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/75dfee071ab560e8309157a7e5222bd9/logTestTxnWithExecutorDataDriven1241454943
    test_log_scope.go:79: use -show-logs to present logs inline
=== RUN   TestTxnWithExecutorDataDriven/basic_txn
    txn_with_executor_datadriven_test.go:52: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/7281/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/catalog/descs/descs_test_/descs_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/catalog/descs/testdata/basic_txn:1:
        exec [1 args]
        CREATE TABLE foo (i INT PRIMARY KEY);
        ----
        0	
    txn_with_executor_datadriven_test.go:52: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/7281/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/catalog/descs/descs_test_/descs_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/catalog/descs/testdata/basic_txn:6:
        exec [1 args]
        INSERT INTO foo VALUES (1), (2);
        ALTER TABLE foo ADD COLUMN j INT DEFAULT 52;
        ALTER TABLE foo ADD COLUMN k INT DEFAULT 64;
        ----
        2	0	0	
    txn_with_executor_datadriven_test.go:52: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/7281/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/catalog/descs/descs_test_/descs_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/catalog/descs/testdata/basic_txn:13:
        query [1 args]
        SELECT * FROM foo;
        SELECT j, k FROM foo;
        ----
        i  j   k
        1  52  64
        2  52  64
        j   k
        52  64
        52  64
=== RUN   TestTxnWithExecutorDataDriven/drop_and_alter_in_txn
    txn_with_executor_datadriven_test.go:52: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/7281/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/catalog/descs/descs_test_/descs_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/catalog/descs/testdata/drop_and_alter_in_txn:1:
        exec [1 args]
        CREATE DATABASE mydb;
        ----
        0	
    txn_with_executor_datadriven_test.go:52: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/7281/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/catalog/descs/descs_test_/descs_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/catalog/descs/testdata/drop_and_alter_in_txn:6:
        exec [1 args]
        ALTER ROLE ALL IN DATABASE mydb SET timezone = 'America/New_York';
        DROP DATABASE mydb;
        ----
        0	0	
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-schema
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTxnWithExecutorDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-25910

sql/catalog/descs.TestTxnWithExecutorDataDriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9323345?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9323345?buildTab=artifacts#/) on master @ [7d796b17bf1b9a18a925ecc33ed1656fece987ed](https://github.com/cockroachdb/cockroach/commits/7d796b17bf1b9a18a925ecc33ed1656fece987ed):

Fatal error:

```
panic: test timed out after 55s

```
Stack: 

```
goroutine 36990 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2036 +0x88
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x38
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestTxnWithExecutorDataDriven
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/75dfee071ab560e8309157a7e5222bd9/logTestTxnWithExecutorDataDriven2879264932
    test_log_scope.go:79: use -show-logs to present logs inline
=== RUN   TestTxnWithExecutorDataDriven/basic_txn
    txn_with_executor_datadriven_test.go:52: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/6048/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/catalog/descs/descs_test_/descs_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/catalog/descs/testdata/basic_txn:1:
        exec [1 args]
        CREATE TABLE foo (i INT PRIMARY KEY);
        ----
        0	
    txn_with_executor_datadriven_test.go:52: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/6048/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/catalog/descs/descs_test_/descs_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/catalog/descs/testdata/basic_txn:6:
        exec [1 args]
        INSERT INTO foo VALUES (1), (2);
        ALTER TABLE foo ADD COLUMN j INT DEFAULT 52;
        ALTER TABLE foo ADD COLUMN k INT DEFAULT 64;
        ----
        2	0	0	
    txn_with_executor_datadriven_test.go:52: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/6048/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/catalog/descs/descs_test_/descs_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/catalog/descs/testdata/basic_txn:13:
        query [1 args]
        SELECT * FROM foo;
        SELECT j, k FROM foo;
        ----
        i  j   k
        1  52  64
        2  52  64
        j   k
        52  64
        52  64
=== RUN   TestTxnWithExecutorDataDriven/drop_and_alter_in_txn
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTxnWithExecutorDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/catalog/descs/BUILD.bazel
+++ b/pkg/sql/catalog/descs/BUILD.bazel
@@ -77,7 +77,7 @@ go_library(
 
 go_test(
     name = "descs_test",
-    size = "small",
+    size = "medium",
     srcs = [
         "collection_test.go",
         "errors_test.go",
@@ -85,7 +85,7 @@ go_test(
         "txn_external_test.go",
         "txn_with_executor_datadriven_test.go",
     ],
-    args = ["-test.timeout=55s"],
+    args = ["-test.timeout=295s"],
     data = glob(["testdata/**"]),
     embed = [":descs"],
     deps = [

